pkg/collector/remote_config.go

```diff
@@ -0,0 +1,88 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package collector
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-agent/pkg/secrets"
+	pkglog "github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// RemoteConfigScheduler is the structure used to run checks with RC
+type RemoteConfigScheduler struct {
+	scheduler     *CheckScheduler
+	runningChecks []integration.Config
+}
+
+type agentIntegration struct {
+	Name       string            `json:"name"`
+	Instances  []json.RawMessage `json:"instances"`
+	InitConfig json.RawMessage   `json:"init_config"`
+}
+
+// secretsDecrypt allows tests to intercept calls to secrets.Decrypt.
+var secretsDecrypt = secrets.Decrypt
+
+// NewRemoteConfigScheduler creates an instance of a remote config integration scheduler
+func NewRemoteConfigScheduler() *RemoteConfigScheduler {
+	return &RemoteConfigScheduler{
+		runningChecks: make([]integration.Config, 0),
+	}
+}
+
+// Start creates the remote-config scheduler
+func (sc *RemoteConfigScheduler) Start(scheduler *CheckScheduler) error {
+	if sc.scheduler != nil {
+		return fmt.Errorf("Remote-config scheduler is already initiated")
+	}
+
+	sc.scheduler = scheduler
+	return nil
+}
+
+// IntegrationScheduleCallback is called at every AGENT_INTEGRATIONS to schedule/unschedule integrations
+func (sc *RemoteConfigScheduler) IntegrationScheduleCallback(updates map[string]state.RawConfig) {
```

Comment:
Ah, we could do it smarter if we had the difference between the 2 configurations.
Can't we get that info from the scheduler? But we would also need to be able to compare the configurations they were scheduled with

Reply:
We can be smarter, but we need to do that carefully and make sure we support configs that are updated
For the first implementation I chose to go with something naive that cover the use cases, do you think we should still be smarter here?